src/Promise.php

```diff
@@ -263,10 +263,17 @@ private function invokeWaitList()
         $this->waitList = null;
 
         foreach ($waitList as $result) {
-            $result->waitIfPending();
-            while ($result->result instanceof Promise) {
-                $result = $result->result;
+            while (true) {
                 $result->waitIfPending();
+
+                if ($result->result instanceof Promise) {
+                    $result = $result->result;
+                } else {
+                    if ($result->result instanceof PromiseInterface) {
+                        $result->result->wait(false);
```

Comment:
I'm a little rust here. Why assume `false`? Should we now pass the `$unwrap` variable from `wait` to this method so that we can unwrap if requested?